src/styles/blocks/calendar.scss

```
@@ -0,0 +1,80 @@
+@function week-width($day-width, $gap-between-days) {
+  @return $day-width * 7 + $gap-between-days * 6 + 20px;
+}
+
+$week-days:
+  "mon" 0,
+  "tue" 1,
+  "wed" 2,
+  "thu" 3,
+  "fri" 4,
+  "sat" 5,
+  "sun" 6;
+
+.calendar {
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  justify-content: flex-start;
+  align-items: center;
+  gap: 1px;
+  width: week-width($day-width, $gap-between-days);
+  padding: 110px 10px 0;
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: $day-width;
+    height: $day-height;
+    background-color: $day-color;
+    border: 1px solid $border-color;
+    transition: transform $transition-duration ease;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        content: "#{$i}";
+        color: $border-color;
+        font-size: $font-size;
+      }
+    }
+
+    @each $start-day, $week-day in $week-days {
+      &--start-day-#{$week-day} &__day:first-child {
+        margin-left: $week-day * ($day-width + $gap-between-days);
+      }
+    }
+
+    @for $i from 28 through 31 {
+      &--month-length-#{$i} {
+        &__day:nth-child(n + #{$i}) ~&__day {
+          display: none;
+        }
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: $hover-day-color;
+      transition: $transition-duration;
+      transform: translate(0, -20px);
+    }
+  }
+
+  @each $start-day, $week-day in $week-days {
+    &--start-day-#{$week-day} &__day:first-child {
+      margin-left: $week-day * ($day-width + $gap-between-days);
+    }
+  }
```

There is a typo here, we want to use `$start-day` in class name. And also to make it more BEM-like I change the class name a bit:
```suggestion
  @each $start-day, $week-day in $week-days {
    &__start-day--#{$start-day} &__day:first-child {
      margin-left: $week-day * ($day-width + $gap-between-days);
    }
  }
```